src/postgres/inventory/collection/PgCollection.ts

```diff
@@ -133,23 +132,35 @@ class PgCollection implements Collection<PgClassType.Value> {
       async (values: Array<PgClassType.Value>): Promise<Array<PgClassType.Value>> => {
         const insertionIdentifier = Symbol()
 
+        // Get the fields that actually have been used in any of the values, so
+        // we can leave the rest out of the query completely to prevent
+        // permission issues
+        const fields = new Map()
+        values.forEach((map) => {
+          map.forEach((value, key) => {
+            if (typeof value !== 'undefined') {
+              fields.set(key, this.type.fields.get(key))
+            }
```

Comment:
My one concern is that this is `O(vn)` where `v` is the number of values and `n` is the maximum number of fields in one of the values. Is there a more efficient algorithm for this? Can we do some other work while iterating? Think about these questions only for a bit, if you can’t think of a good answer I’m comfortable with merging this as an initial implementation.

Ideally I think we could take a similar approach here as we are taking in https://github.com/calebmer/postgraphql/pull/342, although this is a good first implementation. If you could put a `TODO` comment here that reads something like: `TODO: Use field information from GraphQL instead of iterating through values` we can address it when it becomes a problem 👍